src/modules/auth/actions.js

```diff
@@ -63,6 +70,11 @@ export const fetchUserDependentData = (servicesCb) => async (dispatch, getState)
                 () => dispatch(fetchServiceDependentData())));
             await (servicesCb || nop)();
             await dispatch(fetchProjectsWithDatasets());  // TODO: If needed, remove if !hasAttempted
+            const state = getState();
+            const datasetsByID = getDatasetsByID(state);
+            for (const datasetIdentifier of Object.keys(datasetsByID)) {
+                await dispatch(fetchDatasetDataTypesSummaryIfPossible(datasetIdentifier));
```

Comment:
this should be a parallel `Promise.all` for all datasets - they can be done in parallel rather than sequentially

Reply:
You are right; a better way!